server/ocsp.go

```diff
@@ -400,11 +417,52 @@ func (srv *Server) NewOCSPMonitor(tc *tls.Config) (*tls.Config, *OCSPMonitor, er
 			}, nil
 		}
 
-		// GetClientCertificate returns a certificate that's presented to a
-		// server.
-		tc.GetClientCertificate = func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			return &cert, nil
+		// Check whether need to verify staples from a client connection depending on the type.
+		switch kind {
+		case typeStringMap[ROUTER], typeStringMap[GATEWAY], typeStringMap[LEAF]:
+			tc.VerifyConnection = func(s tls.ConnectionState) error {
```

Comment:
This is so that the route client connection can verify that the server has a valid OCSP Staple.

Reply:
LGTM: OCSP staple verification is why `.VerifyConnection` was added as a field member.  See `https://github.com/golang/go/issues/36736` for more.